### PR TITLE
WIP: Converted the QuickNote to use CKEditor

### PR DESCRIPTION
**Pull request by ndickerson**
##### **Description**

Swapped out the plain text editor used in QuickNote for a right text editor

##### **What did you change?**

Replaced the old mechanism for using (sort of) rich text QuickNotes with CKEditor.

##### **Reviewers**
* @jgodi

### CODE REVIEW

File: src/elements/quick-note/QuickNote.ts [hunk 1]
```diff
@@ -16,62 +14,51 @@ const QUICK_NOTE_VALUE_ACCESSOR = {
...
-                      (focus)="onFocus($event)"
-                      (blur)="onTouched($event)"
-                      #textArea>
-            </textarea>
-            <div class="quick-note-overlay"
-                [innerHTML]="formattedNote"
-                [attr.placeholder]="placeholder"
-                 #quickNoteOverlay>
-            </div>
+            <textarea #host></textarea>
```
Comment (jgodi) [inline src/elements/quick-note/QuickNote.ts]: Can this just use the CKEditor component that we already have? Would clean up some of the class?

**Comment by jgodi** [comment]
Can we get focus into the CKeditor when they select an item? 

zxvj40ekpa

**Comment by jgodi** [comment]
The CustomResultsTemplate demo is broken too, you are probably aware just documenting it.

`No component factory found for CustomQuickNoteResults`

File: src/elements/quick-note/QuickNote.ts (hunk 1 above)
Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: Will look into this.

Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: Will look into this.

Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: Will look into this.

**Comment by ndickerson** [comment]
Stripping links has been removed in my corresponding novo branch. 

CustomQuickNoteResults is broken on master

Will test the back spacing more thoroughly. 

We probably don't need cleanup of removed references because while that behavior is possible in novo elements today, we don't remove references in novo. They stick around even after the mention is gone.

**Comment by jgodi** [comment]
Ah, it does. We really need E2E tests in here to catch these things.

I think the clean up of references is important, since this is an open source component and should work even though that Novo doesn't use it in the same way. Or this element should be removed from open source. Otherwise they can backspace just a few characters, keep the link in there and the reference would persist.

**Pull request closed without merging**